www/md_base/src/icons/gear-outline.svg

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>

```

Comment:
Directly use font is OK.

But your choice will be limited to the font collection you have choose and
it won't be quite convenient to merge icons from different sets. There are
also some other shortcomings of icon font such as if the icon is small, it
will blur as the browser is optimizing them as a font. There are also many
problems about hinting. My preference for SVG icons mostly comes from this
article of CSS Tricks https://css-tricks.com/icon-fonts-vs-svg/  It
describes why we should use SVG.

I think if we are using ionicons(http://ionicons.com). The size of bower
won't be a problem.  The repo contains no png icons.

I think if the number of icons we are using is not so many (less than 10
icons), put the icons directly in the repo is ok as long as the license
permit. And we might need to make a vector logo of Buildbot instead of
current bitmap icon in the future, then SVG will be a better choice.


On Fri, Apr 10, 2015 at 8:39 PM, Pierre Tardy <notifications@github.com>
wrote:

> In www/md_base/src/icons/gear-outline.svg
> <https://github.com/buildbot/buildbot/pull/1626#discussion_r28140864>:
>
> > @@ -0,0 +1,40 @@
> > +<?xml version="1.0" encoding="utf-8"?>
>
> There is also
> http://zavoloklom.github.io/material-design-iconic-font/icons.html
>
> Problem of using the official bower package for material-design-icons is
> that it downloads the full set of png version, which is a painful 50MB.
>
> The full font is 50kb compressed, so I dont thing this is worth the pain
> of concatenating the svgs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/buildbot/buildbot/pull/1626/files#r28140864>.
>